src/hermes/commands/deposit/file.py

```diff
@@ -7,24 +7,19 @@
 
 import json
 
-import click
-
 from hermes import config
-from hermes.model.context import CodeMetaContext
+from hermes.commands.deposit.base import BaseDepositPlugin
 from hermes.model.path import ContextPath
 
 
-def dummy_noop(click_ctx: click.Context, ctx: CodeMetaContext):
-    pass
-
-
-def map_metadata(click_ctx: click.Context, ctx: CodeMetaContext):
-    ctx.update(ContextPath.parse('deposit.file'), ctx['codemeta'])
+class FileDepositPlugin(BaseDepositPlugin):
+    def map(self) -> None:
+        self.ctx.update(ContextPath.parse('deposit.file'), self.ctx['codemeta'])
```

Comment:
**thought:** This might as well be the default behaviour (i.e., moved up to base class)

Reply:
I was hesitant to give any of the actual steps default functionality because I didn't want to force people to think about whether (and when) they need to call `super().step()` in their implementation.